### PR TITLE
Fixed: Properties window opens behind main app window

### LINKED ISSUE #9926: Bug: Properties window opens behind main app window
**Steps to reproduce**
1. Right click file or folder
2. Click on properties
3. Observe that the properties window opens behind the main app window

**Requirements**
- Property window should be opened in the foreground

**Version information**
Version: 2.3.26.0
OS Version: 10.0.25193.1000

### PR DESCRIPTION
**Pull request by ferrariofilippo**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #9926 

**Comments**
- The root of the problem was that, after the properties windows was opened, the focus returned to the main window because of the closing of the flyout context menu. I fixed the bug opening Properties after the menu gets closed
- If everything is ok, I'll proceed with code cleanup

**Validation**
How did you test these changes?
- [x] Built and ran the app

### CODE REVIEW

**Comment by gave92** [comment]
There is also a simpler alternative:
```cs
// Workaround for #9926
propertiesWindow.IsAlwaysOnTop = true;
propertiesWindow.DispatcherQueue.CreateTimer().Debounce(() =>
{
    propertiesWindow.Activate();
    propertiesWindow.IsAlwaysOnTop = false;
}, TimeSpan.FromSeconds(1));
```
Adding the above in OpenPropertiesWindowAsync() (@ line 82) appears to be enough to fix the issue.

**Comment by ferrariofilippo** [comment]
> To be noted that there is also a simpler alternative:
> 
> ```cs
> // Workaround for #9926
> propertiesWindow.IsAlwaysOnTop = true;
> propertiesWindow.DispatcherQueue.CreateTimer().Debounce(() =>
> {
>     propertiesWindow.Activate();
>     propertiesWindow.IsAlwaysOnTop = false;
> }, TimeSpan.FromSeconds(1));
> ```
> 
> Adding the above in OpenPropertiesWindowAsync() (@ line 82) appears to be enough to fix the issue. It's a bit hacky though.

I took into account a solution like this, but there are two drawbacks: if, for any reason, the flyout context menu takes more than a second to close, the bug would remain unfixed (it would be a one-in-a-million case). The second problem is that the program might seem not optimized or inefficient because of that delay. I know it's only a second, but I'm obsessed with optimization.

**Comment by gave92** [comment]
For "optimization" the simple solution is actually better since you don't wait for the menu to close before opening the window. I agree that it might not cover all cases (I'm not a fan of debounce times).

**Comment by yaichenbaum** [comment]
Using the hwnd, can we force the new window to the foreground?

**Comment by gave92** [comment]
With my solution I set it always on top for 1 second. It appears in front but it's not active. Not sure if we can force it to be active.

**Comment by ferrariofilippo** [comment]
> Using the hwnd, can we force the new window to the foreground?

You can't really do that the way we'd want to.
Setting `IsAlwaysOnTop` to true means that the Properties Window can't be hidden by any other window (I don't think we want this).
Forcing the Prop. Window to be shown above others (using `BringToFront()` or `MoveInZOrderToTop()`) would be useless, because, as the context menu closes, the main window will be brought on top of others.
Mine and @gave92 solutions are the only viable.

File: src/Files.App/IBaseLayout.cs [hunk 1]
```diff
@@ -25,5 +25,6 @@ public interface IBaseLayout : IDisposable
         public SelectedItemsPropertiesViewModel SelectedItemsPropertiesViewModel { get; }
         public DirectoryPropertiesViewModel DirectoryPropertiesViewModel { get; }
         public BaseLayoutCommandsViewModel CommandsViewModel { get; }
+        public Microsoft.UI.Xaml.Controls.CommandBarFlyout ItemContextMenuFlyout { get; set; }
```
Comment (yaichenbaum) [inline src/Files.App/IBaseLayout.cs:28]: Can this be simplified?

**Review by yaichenbaum — APPROVED** [review]

**Review by yaichenbaum — APPROVED** [review]
LGTM

**Comment by yaichenbaum** [comment]
@ferrariofilippo does this work for the drives widget as well?

**Comment by ferrariofilippo** [comment]
> @ferrariofilippo does this work for the drives widget as well?

Yes, it does

**Comment by gave92** [comment]
@ferrariofilippo solution looks good to me. I just find it annoying that it has to be done in each place we have a context menu and each have slightly different code, but no biggie.

**Comment by yaichenbaum** [comment]
Likewise but this is still a good solution

**Pull request merged**